src/request.rs

```
@@ -351,7 +351,12 @@ pub struct LocalScope<'a> {
 impl<'a> Drop for LocalScope<'a> {
     fn drop(&mut self) {
         if self.num_requests.get() != 0 {
-            panic!("at least one request was dropped without being completed");
+            std::mem::forget(std::panic::catch_unwind(|| {
```

Why `panic!`, `catch_unwind` and `forget` instead of just `eprintln!`? Is this to produce a stack trace? And if so, why would
```
let _ = std::panic::catch_unwind(|| ...);
```
not suffice to ignore the result? Why must it be leaked?

---

Yeah, that's why I wanted to keep the `panic!` - to maintain context.

You're right, I probably can use `let _ = ` - I'll fix that.

---

One thing I am still considering is code density again. IIRC, some places in the standard library do plain `abort` to avoid introducing jumps into the panic machinery in hot code parts.

Considering context: `SIGABRT` should stop my debugger right at the call to `abort` and allow me to debug the issue, so personally, I could live without the panic here.

If you do keep it, it might be useful to put the how `panic`, `catch_unwind`, `abort` sequence into a separate function and mark that `#[cold]` or at least `#[inline(never)]` to avoid putting those landing pads everywhere `LocalScope`'s `Drop` implementation is inlined.

---

(Just for reference, I think I was remembering places like https://github.com/rust-lang/rust/blob/08deb863bdebfcbbb71c18acf903eca84f1df4e7/library/alloc/src/sync.rs#L1174 and https://github.com/rust-lang/rust/blob/2d6cbd21b2c5819c7fa42eb5a3713667b67e4f03/library/alloc/src/rc.rs#L2107)